### PR TITLE
Scope pending operational certificates to their owning fabric index

### PR DESCRIPTION
**Pull request by Alami-Amine**
##### Problem

-   `PersistentStorageOpCertStore` holds **one** set of pending certificates for the whole node, owned by a single fabric index recorded in `mPendingFabricIndex`.
-   `HasAnyCertificateForFabric` tested whether *any* pending certificate existed without comparing the requested index against that member, so `RemoveOpCertsForFabric` accepted an index owning no state of its own and then called the index-free `RevertPendingOpCerts`.
-   Removing the certificates of one fabric therefore discarded a pending chain staged by a **different** fabric:
    -   a concurrent `AddNOC` fails with `kInvalidNOC`, which points at the certificate the caller supplied rather than at the missing pending state
    -   a concurrent `UpdateNOC` commits part-way, and the error epilogue in `CommitPendingFabricData` then removes the fabric that was being updated

##### Solution

-   Compare the requested index against `mPendingFabricIndex` in the pending term, and revert pending certificates only for the index that owns them. Both conditions are required: an index holding its own persisted certificates satisfies the first on the storage lookups alone.
-   Add the missing pending-NOC-chain condition to the update pre-flight in `FabricTable::CommitPendingFabricData`, mirroring the trusted-root check already performed on the add path. An incoherent update now fails before the commit transaction touches storage rather than during it.

##### Caveats

-   Behaviour change on last-fabric removal: a pending trusted root staged under another administrator's fail-safe now survives it, where previously it was discarded as a side effect. It is still cleared when that fail-safe resolves. `TrustedRootCertificates` publishes pending roots, so the difference is observable.

#### Testing

-   Unit tests covering both paths follow in a separate PR.
-   Verified locally that the existing suites are unaffected: `TestFabricTable` 31/31 and `TestPersistentStorageOpCertStore` 5/5, built with `is_asan=true is_clang=true`.

### CODE REVIEW

**Comment by Alami-Amine** [comment]
@coderabbitai full review

File: src/python_testing/TestFabricPendingCertScope.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+from matter.testing.matter_testing import MatterBaseTest
+from matter.testing.runner import default_matter_test_main
+from matter.utils import CommissioningBuildingBlocks
+
+log = logging.getLogger(__name__)
+
+# An index in the valid 1..254 range that holds no fabric on a freshly commissioned DUT.
+UNOCCUPIED_FABRIC_INDEX = 200
+
+FAILSAFE_SECONDS = 15
```
[the remark this answers is not in the document]
Comment (Alami-Amine) [reply src/python_testing/TestFabricPendingCertScope.py]: @coderabbitai I Raised it to 60 s for both AddNoc and UpdateNOC flows, 600s is too long

**Comment by Alami-Amine** [comment]
@coderabbitai full review

File: src/credentials/tests/TestOpCertStorePendingFabricScope.cpp [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+    MutableByteSpan certSpan{ certBuf };
+    EXPECT_SUCCESS(fabricTableHolder.GetOpCertStore().GetCertificate(updatedIndex, CertChainElement::kNoc, certSpan));
+}
+
+// The destructive case: a second fabric is genuinely removed while a commissioned fabric has an
+// UpdateNOC in flight. Before the pending set was scoped to its owning index, the removal cleared
+// the updating fabric's pending certificates while FabricTable kept its pending flags, so
+// CommitPendingFabricData passed every pre-flight, committed the metadata, then failed in
+// CommitOpCertsForFabric and deleted the already-commissioned fabric.
+TEST_F(TestOpCertStorePendingFabricScope, DeleteOfOccupiedIndexDoesNotDisturbFabricUnderUpdate)
```
Comment (cecille) [inline src/credentials/tests/TestOpCertStorePendingFabricScope.cpp:218]: A bit orthogonal to this PR, but does everything behave OK if the fabric currently being updated gets removed?

File: src/python_testing/TestFabricPendingCertScope.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+log = logging.getLogger(__name__)
+
+# An index in the valid 1..254 range that holds no fabric on a freshly commissioned DUT.
+UNOCCUPIED_FABRIC_INDEX = 200
+
+FAILSAFE_SECONDS = 60
+
+
+class TestFabricPendingCertScope(MatterBaseTest):
```
Comment (cecille) [inline src/python_testing/TestFabricPendingCertScope.py:64]: This isn't currently set up as a certification test - should it be? Older devices will fail.

**Review by cecille — APPROVED** [review]

File: src/python_testing/TestFabricPendingCertScope.py (hunk 3 above)
Comment (tcarmelveilleux) [reply src/python_testing/TestFabricPendingCertScope.py]: I don't think it should be a cert test. This bug is a big corner case and I don't think it requires separate spec update. It would be good to have a comment though.

File: src/credentials/PersistentStorageOpCertStore.cpp [hunk 4]
```diff
@@ -541,7 +541,10 @@ bool PersistentStorageOpCertStore::HasAnyCertificateForFabric(FabricIndex fabric
     bool rcacMissing = !StorageHasCertificate(mStorage, fabricIndex, CertChainElement::kRcac);
     bool icacMissing = !StorageHasCertificate(mStorage, fabricIndex, CertChainElement::kIcac);
     bool nocMissing  = !StorageHasCertificate(mStorage, fabricIndex, CertChainElement::kNoc);
-    bool anyPending  = (mPendingRcac.Get() != nullptr) || (mPendingIcac.Get() != nullptr) || (mPendingNoc.Get() != nullptr);
+    // Pending VID verification elements need no term here: they are only staged while
+    // mPendingFabricIndex is set, which in turn only happens alongside a pending RCAC or NOC.
+    bool anyPending = (fabricIndex == mPendingFabricIndex) &&
```
Comment (tcarmelveilleux) [inline src/credentials/PersistentStorageOpCertStore.cpp]: Suggest "anyPendingForRequestedFabric"

File: src/credentials/tests/TestOpCertStorePendingFabricScope.cpp [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+    const FabricInfo * fabricInfo = fabricTable.FindFabricWithIndex(updatedIndex);
+    ASSERT_NE(fabricInfo, nullptr);
+    EXPECT_EQ(fabricInfo->GetNodeId(), kNodeIdAfter);
+
+    uint8_t certBuf[kMaxCHIPCertLength];
+    MutableByteSpan certSpan{ certBuf };
+    EXPECT_SUCCESS(fabricTableHolder.GetOpCertStore().GetCertificate(updatedIndex, CertChainElement::kNoc, certSpan));
+}
+
+// The destructive case: a second fabric is genuinely removed while a commissioned fabric has an
```
Comment (tcarmelveilleux) [inline src/credentials/tests/TestOpCertStorePendingFabricScope.cpp]: "the destructive case" is specific to context for an agent at one point in time. Please reword the comments to stand on their own.

**Review by tcarmelveilleux — APPROVED** [review]
